### PR TITLE
First Draft of Code. Debug/Testing left

### PR DESCRIPTION
**Pull request by BigBang0072**
Please comment on the overall format of the code. I will complete the testing by tomorrow.

### CODE REVIEW

File: GSOC18/hexCells_to_squareCell_interpolation.py [hunk 1]
```diff
@@ -0,0 +1,162 @@
...
+from scipy.spatial import  cKDTree
+#Plotting Imports and configuration
+import matplotlib.pyplot as plt
+#Importing custom classes and function
+from sq_Cells import sq_Cells
+
+
+#################Global Variables#######################
+dtype=np.float64            #data type of any numpy array created
+sq_cells_filename='/home/abhinav/Desktop/HAhRD/GSOC18/sq_cells_data/sq_cells_dict.pkl'
```
Comment (BigBang0072) [inline GSOC18/hexCells_to_squareCell_interpolation.py]: Please change the relative path according to system

File: GSOC18/hexCells_to_squareCell_interpolation.py [hunk 2]
```diff
@@ -0,0 +1,162 @@
...
+    #Time Comlexity = O(res[0]*res[1])
+    for i in range(resolution[0]):
+        for j in range(resolution[1]):
+            #Center of the square polygon
+            center=(min_x+i*x_length,min_y+j*y_length)
+            id=(i,j)    #given in usual matrix notation
+            sq_cells[id]=sq_Cells(id,center,x_length,y_length)
+
+    #Saving the sq_cell sq_cell_data in given folder
+    fhandle=open(sq_cells_filename,'wb')
```
Comment (BigBang0072) [inline GSOC18/hexCells_to_squareCell_interpolation.py]: The pickle file is being saved here for the square cells which I mentioned in the mail, according to filename given in "global variable" section of this file.

File: GSOC18/main.py [hunk 3]
```diff
@@ -0,0 +1,80 @@
+from hexCells_to_squareCell_interpolation import linear_interpolate_hex_to_square
+import sys
+sys.path.append('/home/abhinav/Desktop/HAhRD/GSOC18/hex_cells_data/cells_out.pkl')
```
Comment (BigBang0072) [inline GSOC18/main.py]: Please ignore this line. I don't know why it is not able to add the path. Also, the directory structure is changed from here.

File: GSOC18/main.py [hunk 4]
```diff
@@ -0,0 +1,80 @@
...
+        #ax1.set_ylim(-160, 160)
+        #ax1.set_aspect(1)
+        plt.show()
+
+
+
+if __name__=='__main__':
+
+    ## Generating the overlapping coefficient
+    hex_filename='/home/abhinav/Desktop/HAhRD/GSOC18/hex_cells_data/hex_cells_dict.pkl'
```
Comment (BigBang0072) [inline GSOC18/main.py]: From here the hexagonal cells dictionary will be read. So, they need to be saved in this location before using.

File: GSOC18/main.py [hunk 5]
```diff
@@ -0,0 +1,80 @@
...
+    fhandle=open(hex_filename,'rb')
+    hex_cells_dict=pickle.load(fhandle)
+    fhandle.close()
+    resolution=(100,100)
+    layer=1
+    sq_coef=linear_interpolate_hex_to_square(hex_filename,
+                                                layer,resolution)
+
+    ## Plotting the sq cell for verification
+    sq_filename='/home/abhinav/Desktop/HAhRD/GSOC18/sq_cells_data/sq_cells_dict.pkl'
```
Comment (BigBang0072) [inline GSOC18/main.py]: This is automatically saved by the script. Just the pathname should be given in the previous interpolate_hexCell... script.

File: GSOC18/sq_Cells.py [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+        #Adding a Point object as the center of the square cells
+        self.center=Point(center)
+
+        #Now creating a Polygon(box) object as the square cells
+        minx=center[0]-x_length/2
+        maxx=center[0]+x_length/2
+        miny=center[1]-y_length/2
+        maxy=center[1]+y_length/2
+        self.polygon=box(minx,miny,maxx,maxy)
+        #should we keep the name polygon or vertices as in
```
Comment (BigBang0072) [inline GSOC18/sq_Cells.py:24]: Please see the comment here.

**Review by grasseau — APPROVED** [review]
Seems Ok. I need to have a global view and to run the code. Thanks

**Review by grasseau — APPROVED** [review]
Ok

**Comment by grasseau** [comment]
First merge request ... test

**Pull request merged**